src/status_im/contexts/wallet/create_account/new_keypair/keypair_name/view.cljs

```diff
@@ -42,4 +50,20 @@
             :label           (i18n/label :t/keypair-name)
             :char-limit      keypair-name-max-length
             :auto-focus      true
-            :on-change-text  #(reset! keypair-name %)}]]]))))
+            :on-change-text  (fn [value]
+                               (reset! keypair-name value)
+                               (cond
+                                 (> (count value) keypair-name-max-length) (set-error? 1)
+                                 (has-emojis? value)                       (set-error? 2)
+                                 (has-special-characters? value)           (set-error? 3)
+                                 :else                                     (set-error? nil)))
+            :error?          error?}]
+          (when error?
+            [rn/view
+             {:style style/error-container}
+             [quo/icon :i/info {:color (colors/theme-colors colors/danger-50 colors/danger-60 theme)}]
+             [quo/text
+              {:style {:margin-left 4
+                       :color       (colors/theme-colors colors/danger-50 colors/danger-60 theme)}}
+              (i18n/label (keyword (str "t/key-name-error-" error?)))]])]]))))
```

Comment:
imo it would just be nicer to have a map with the errors resolve individually.
```
(def error-messages {
    1 (i18n/label :t/key-name-error-1)
    2 (i18n/label :t/key-name-error-2)
    3 (i18n/label :t/key-name-error-3)
}
```
as it reads a bit more naturally and not so sure about this keyword/string resolving approach here.